### PR TITLE
Cleanup search form

### PR DESCRIPTION
**Pull request by rcantin-w**
## Who is this for?
Maintenance

## What is it doing for them?
As we're [killing the old search](https://github.com/wellcomecollection/wellcomecollection.org/pull/9371), I wanted to make sure we removed traces of the old `Sort`. I know this kind of plays into https://github.com/wellcomecollection/wellcomecollection.org/issues/9351 as well, but I'm hoping it just helps the refactor later on, having removed props that were no longer used/needed + used the former UI look.

Part of #9034

### CODE REVIEW

File: common/views/components/SearchForm/SearchForm.SortByPortal.tsx [hunk 1]
```diff
@@ -1,26 +0,0 @@
-import { FunctionComponent, ReactElement, useEffect, useState } from 'react';
-import { createPortal } from 'react-dom';
```
Comment (alexwlchan) [inline common/views/components/SearchForm/SearchForm.SortByPortal.tsx:2]: It looks like this is the last use of `react-dom` in the project, so let's remove it from `package.json` also.

File: common/views/components/SearchForm/SearchForm.tsx [hunk 2]
```diff
@@ -14,17 +13,11 @@ import { ParsedUrlQuery } from 'querystring';
 import TextInput from '@weco/common/views/components/TextInput/TextInput';
 import ButtonSolid from '@weco/common/views/components/ButtonSolid/ButtonSolid';
 import { trackGaEvent } from '@weco/common/utils/ga';
-import SearchFilters from '@weco/common/views/components/SearchFilters';
-import Select from '@weco/common/views/components/Select/Select';
 import Space from '@weco/common/views/components/styled/Space';
-import SelectUncontrolled from '@weco/common/views/components/SelectUncontrolled/SelectUncontrolled';
```
Comment (alexwlchan) [inline common/views/components/SearchForm/SearchForm.tsx:20]: Looks like SelectUncontrolled isn't being used anywhere else, should we clean it up while we're here?

**Review by alexwlchan — CHANGES_REQUESTED** [review]
Looks good, but remember to clean up the dependencies too.

File: common/views/components/SearchForm/SearchForm.SortByPortal.tsx (hunk 1 above)
Comment (rcantin-w) [reply common/views/components/SearchForm/SearchForm.SortByPortal.tsx:2]: I believe it's part of Next's build setup and is needed to run the project in general (paired with `react`). It won't build without. But good shout in checking dependencies, I'll have a look through!

**Review by alexwlchan — APPROVED** [review]

**Pull request merged**